File: client/src/schemas/task.ts
import {z} from "zod/v4-mini";

let PriorityEnum = z.enum(["LOW", "MEDIUM", "HIGH", "VITAL"]);
export let TaskSchema = z.object({
	id: z.number(),
	title: z.string(),
	description: z.string(),
	priority: PriorityEnum,
	completed: z.boolean(),
});

export let CreateTaskSchema = z.object({
	title: z.string(),
	description: z.string(),
	priority: PriorityEnum,
	completed: z.boolean(),
});

export type Priority = z.infer<typeof PriorityEnum>;
export type Task = z.infer<typeof TaskSchema>;
export type CreateTask = z.infer<typeof CreateTaskSchema>;

File: server/ktor-exposed-db-app/src/main/kotlin/dto/Task.kt
package com.example.dto

import kotlinx.serialization.Serializable


enum class Priority {
  HIGH,
  MEDIUM,
  LOW,
  VITAL
}

@Serializable
data class Task(
  val id: Int,
  val title: String,
  val description: String,
  val priority: Priority,
  val completed: Boolean = false,
)


@Serializable
data class CreateTask(
  val title: String,
  val description: String,
  val priority: Priority = Priority.MEDIUM,
  val completed: Boolean = false,
)